src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,197 @@
+            tobeMark.setMark();
+           return "You could have gotten me to help you. This task has been marked done" +
+                   tobeMark.toString();
+        }
+    }
+
+    /**
+     * Add event type task to tasklist
+     *
+     * @param command command by user which specify task description
+     * @param storeList tasklist to be added to
+     * @return String of response
+     *
+     */
+    public static String addEvent(String command, ArrayList<Task> storeList) {
+        int slash = command.indexOf("/");
+        if (slash == -1 || slash <= 6) {
+            return "Master, you wished wrongly. Remember you have to wish in this format " +
+                    "deadline task /at dateofevent. Please wish again";
+        } else {
+            String newtask = command.substring(6, slash - 1);
+            String attime = command.substring(slash + 1);
+            Event e = new Event(newtask, attime);
+            storeList.add(e);
+            return "Added to my brain master:" + e.toString() + " Currently I have " +
+                    storeList.size() + " things in my brain";
+        }
+    }
+
+    /**
+     * Add Todo type task to tasklist
+     *
+     * @param command command by user which specify task description
+     * @param storeList tasklist to be added to
+     * @return String of response
+     *
+     */
+    public static String addTodo(String command, ArrayList<Task> storeList) {
+        if (command.length() < 5) {
+            return "Master, I have all the knowledge in the world but I do not know what you want to do," +
+                    " Please wish again in the format todo task";
+        } else {
+            String newtask = command.substring(5);
+            Todo t = new Todo(newtask);
+            storeList.add(t);
+            return "Added to my brain master:" + t.toString() + "Currently I have " +
+                    storeList.size() + " things in my brain";
+        }
+    }
+
+    /**
+     * Add Deadline type task to tasklist
+     *
+     * @param command command by user which specify task description
+     * @param storeList tasklist to be added to
+     * @return Nothing
+     *
+     */
+    public static String addDeadline(String command, ArrayList<Task> storeList) {
+        int slash = command.indexOf("/");
+        if (slash == -1 || slash <= 9) {
+            return "Master, you wished wrongly. Remember you have to wish in this format " +
+                    "deadline task /by dateofdeadline. Please wish again";
+        } else {
+            String newtask = command.substring(9, slash - 1);
+            if (newtask.equals("")) {
+                return "Master, you wished wrongly. Remember you have to wish in this format " +
+                        "deadline task /by dateofdeadline. Please wish again";
+            } else {
+                String endtime = command.substring(slash + 1);
+                Deadline d = new Deadline(newtask, endtime);
```

Comment:
how about changing the variable "d" to other names which will give more information about what you are doing :)